pkg/steps/git_source.go

```diff
@@ -23,12 +23,16 @@ func (s *gitSourceStep) Inputs(ctx context.Context, dry bool) (api.InputDefiniti
 }
 
 func (s *gitSourceStep) Run(ctx context.Context, dry bool) error {
+	ref := s.jobSpec.Refs.BaseRef
+	if len(s.jobSpec.Refs.Pulls) > 0 && s.jobSpec.Refs.Pulls[0].Number != 0 {
```

Comment:
what if this is the batch merge where you have multiple pulls? given it's an array

Reply:
for that reason, I get only the first value.  If the batch job will have other PRs that are changing the Dockerfile, they will be ignored.

Reply:
Do you have any other suggestions?

Reply:
clone master, merge all the PRs start build from local dir? or push the batch to some remote temporarily